### DISCUSSION TITLE
Return type of Livewire\TemporaryUploadedFile::getPath() should either be compatible with SplFileInfo::getPath(): string, or the #[\ReturnTypeWillChange] attribute should be used

### DISCUSSION THREAD

**Discussion by goodevilgenius** [livewire/livewire#4419] [7. Bugs (archived)] [answered]
I'm getting this error currently when visiting a user profile page. This is a fairly fresh project. I believe this error appeared after upgrading to PHP 8.1

Here's a little context:

- Controller: Laravel\Jetstream\Http\Controllers\Livewire\UserProfileController@show
- Laravel version: 8.64.0
- PHP version: 8.1.0
- Livewire version: 2.8.2

**Comment by goodevilgenius** [comment]
Adding that attribute revealed the same error for getSize, getFilename, and getRealPath.

**Comment by joshhanley** [comment]
@goodevilgenius can you post a screenshot of the error?

Thanks!

**Comment by goodevilgenius** [comment]
@joshhanley 

![Screenshot_20211224-143641](https://user-images.githubusercontent.com/254662/147371718-2875da83-506e-49f4-bc01-135f7608c2c8.png)

It's a deprecation warning, rather than an actual error. I was able to fix it by turning off deprecation errors.

**Reply by joshhanley** [reply]
@goodevilgenius ah right makes sense! Hmm I'm not sure how we can fix that as we can't use attributes due to supporting php 7.2 and up. Will have to look into it.

Thanks!

**Comment by goodevilgenius** [comment]
@joshhanley Attributes are backwards-compatible. It'll be seen as just a comment in 7.2. Adding them will allow it to be forwards-compatible to 8.1, while staying backwards-compatible with 7.2.

But, I think adding the return type should also fix this error.

**Comment by joshhanley** [comment, marked as answer]
@goodevilgenius thanks I wasn't aware of that!

It turns out my comment here was premature, as there is currently an open PR #4485 to fix this issue 🙂

It uses attributes, as adding return types would be a breaking change.

Hope this helps!
